Bookings: Add empty and error states for booking list

itsmeichigo opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses. -->
<!-- Part of: # -> use this if your PR is one of many related to one issue -->

Closes WOOMOB-1394

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR updates the UI of the booking list when there are no bookings or fetching fails. For the empty state of the All tab without filters, I'm still figuring out the copy with Wagner, so I'm keeping just the generic title and no description for now.

The offline banner has been updated to be added with SwiftUI instead of the UIKit approach as adding subviews directly to hosting controller is not supported.

Additionally, the tab bar is also updated to better adjust when large font sizes are used.

## Testing steps
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->

1. Log in to a CIAB store with existing bookable products.
2. Navigate to the Bookings tab. 
3. Confirm that if the booking list is empty on any tab, the empty state is displayed correctly (matching the design).
4. Disconnect the Internet. 
5. Confirm that an offline banner is displayed, and an error message is displayed as a snack bar at the bottom after fetching fails. Tapping on the snack bar will dismiss it.

## Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->

Tested with simulator iPhone 17.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->

\ | Light mode | Dark mode
--- | --- | ---
Today | <img width="320" alt="Simulator Screenshot - iPhone 17 - 2025-10-06 at 15 32 24" src="https://github.com/user-attachments/assets/d4aae80f-3ad7-4061-8785-be92c00614fa" /> | <img width="320" alt="Simulator Screenshot - iPhone 17 - 2025-10-06 at 15 32 36" src="https://github.com/user-attachments/assets/50d147a5-e34e-4914-923b-d04792f8f87b" /> 
Upcoming | <img width="320" alt="Simulator Screenshot - iPhone 17 - 2025-10-06 at 15 32 45" src="https://github.com/user-attachments/assets/1c0ab0e5-406c-4d58-9603-97fd761c752e" /> | <img width="320" alt="Simulator Screenshot - iPhone 17 - 2025-10-06 at 15 32 53" src="https://github.com/user-attachments/assets/e4b94dc1-5006-44af-a9c3-8c9a269f1d0b" /> 
Filter | <img width="320" alt="Simulator Screenshot - iPhone 17 - 2025-10-06 at 15 37 57" src="https://github.com/user-attachments/assets/bdf21e7e-87d6-4a91-9f7b-d72e6edbbb18" /> | <img width="320" alt="Simulator Screenshot - iPhone 17 - 2025-10-06 at 15 38 04" src="https://github.com/user-attachments/assets/faf3af17-aa0d-4d90-97a7-736cb87e20cc" />
Error | <img width="320" alt="Simulator Screenshot - iPhone 17 - 2025-10-06 at 16 47 16" src="https://github.com/user-attachments/assets/845922cb-ad1e-4967-bb47-48a6277aecae" /> | <img width="320" alt="Simulator Screenshot - iPhone 17 - 2025-10-06 at 16 47 26" src="https://github.com/user-attachments/assets/07d3792b-8a22-40b1-8bab-cedd7381c42a" />

Offline banner:

<img width="320" alt="Simulator Screenshot - iPhone 17 - 2025-10-07 at 08 46 43" src="https://github.com/user-attachments/assets/effef674-a51e-42c5-b90b-947a67acaa25" />

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

itsmeichigo committed:
Add image for empty state

itsmeichigo committed:
Add empty states

itsmeichigo committed:
Enable refreshing empty state

itsmeichigo committed:
Make tab bar adjust to large size fonts

itsmeichigo committed:
Update content size for empty state view

itsmeichigo committed:
Add error snack bar

itsmeichigo committed:
Fix position of the tab indicator

itsmeichigo committed:
Remove unused code

itsmeichigo committed:
Remove description if all bookings are not found

itsmeichigo committed:
Make toast animate

itsmeichigo committed:
Fix offline banner on bookings tab

itsmeichigo committed:
Remove unused code

RafaelKayumov reviewed (commented):
Code looks good. I noticed a couple issues in behaviour:
- The missing internet bar lacks side paddings. It's noticeable on narrow screens. Also the text gets truncated in Russian:

<img width="490" height="292" alt="Снимок экрана 2025-10-08 в 01 19 15" src="https://github.com/user-attachments/assets/835b549c-f457-4e72-ba4b-68faf1257235" />

![photo_2025-10-08_01-23-37](https://github.com/user-attachments/assets/6b19a6a1-7c7b-44e3-aad8-e6f422db38ff)

- When switching internet on/off, the 1st tab is force selected. Check out the video below:

https://github.com/user-attachments/assets/ac1267f1-fd74-45e0-a296-1cbebf77a20b

itsmeichigo committed:
Fix offline banner issues

itsmeichigo committed:
Merge branch 'trunk' into woomob-1394-booking-list-empty-state-and-error-handling

itsmeichigo committed:
Remove redundant view model

itsmeichigo commented:
@RafaelKayumov Thanks for spotting the issues! I added a fix in 5b5fb8e32bbd0236f9e0e4333a8eed83e7d0a046:
- For simplicity, I updated the offline banner to adjust the font size to fit the content, and avoid updating the size when the size category changes.
- Keeping the view model of the list container a `@StateObject` solves the problem. Since you already did that in a previous PR, merging trunk to this branch should be sufficient.

This PR is ready for another look 🙏
